src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityEventSummary.kt

```diff
@@ -49,6 +114,62 @@ object HoppityEventSummary {
         if (event.chocGained > 0) stats.dupeChocolateGained += event.chocGained
     }
 
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        reCheckInventoryState()
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        reCheckInventoryState()
+    }
+
+    @SubscribeEvent
+    fun onKeyPress(event: LorenzKeyPressEvent) {
+        reCheckInventoryState()
+        if (!liveDisplayConfig.enabled) return
+        if (liveDisplayConfig.toggleKeybind == Keyboard.KEY_NONE || liveDisplayConfig.toggleKeybind != event.keyCode) return
+        // Only toggle from inventory if the user is in the Chocolate Factory
+        if (Minecraft.getMinecraft().currentScreen != null && !ChocolateFactoryAPI.inChocolateFactory) return
+        if (lastToggleMark.passedSince() < 250.milliseconds) return
+        val storage = storage ?: return
+        storage.hoppityStatLiveDisplayToggled = !storage.hoppityStatLiveDisplayToggled
+        lastToggleMark = SimpleTimeMark.now()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!HoppityAPI.isHoppityEvent()) return
+        val stats = getYearStats().first ?: return
+
+        if (rabbitTheFishPattern.matches(event.message)) {
+            stats.rabbitTheFishFinds++
+        }
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent) {
+        if (!liveDisplayEnabled()) return
+        val storage = storage ?: return
+        val statYear = storage.hoppityStatLiveDisplayYear.takeIf { it != -1 }
+            ?: SkyBlockTime.now().year
+
+        val stats = getYearStats(statYear).first
+        // Calculate a 'hash' of the stats to determine if they have changed
+        val statsHash = stats?.hashCode() ?: 0
```

Comment:
```suggestion
        val statsHash = stats.hashCode()
```
kotlin already has an inline function that runs hashcode on a nullable and returns 0 if null